Lib/configparser.py

```diff
@@ -561,36 +561,31 @@ def __init__(self):
         self.errors = list()
 
 
-class _Line(str):
+class _LineParser:
 
-    def __new__(cls, val, *args, **kwargs):
-        return super().__new__(cls, val)
-
-    def __init__(self, val, prefixes):
-        self.prefixes = prefixes
-
-    @functools.cached_property
-    def clean(self):
-        return self._strip_full() and self._strip_inline()
+    def __init__(self, comments):
+        self.comments = comments
 
     @property
-    def has_comments(self):
-        return self.strip() != self.clean
+    def value(self):
+        return self._value
 
-    def _strip_inline(self):
-        """
-        Search for the earliest prefix at the beginning of the line or following a space.
-        """
-        matcher = re.compile(
-            '|'.join(fr'(^|\s)({re.escape(prefix)})' for prefix in self.prefixes.inline)
-            # match nothing if no prefixes
-            or '(?!)'
-        )
-        match = matcher.search(self)
-        return self[:match.start() if match else None].strip()
+    @value.setter
+    def value(self, string):
+        self._value = string
+        string = string.strip()
+        self.clean = self._strip_full(string) and self._strip_inline(string)
+        self.has_comments = string != self.clean
+
+    def _strip_full(self, string):
+        return '' if any(map(string.startswith, self.comments.full)) else True
 
-    def _strip_full(self):
-        return '' if any(map(self.strip().startswith, self.prefixes.full)) else True
+    def _strip_inline(self, string):
+        if self.comments.inline:
+            match = self.comments.inline.search(string)
+            if match:
+                return string[:match.start()].rstrip()
+        return string
```

Comment:
My preference instead of writing branching code is to write code that acts unconditionally, such as the previous regular expression, instead of tracking state (self.comments.inline is None) and altering the behavior based on state. I do this because it simplifies the analysis of the logic for correctness (fewer branches and states to track).
I can see how my preferred approach has potential performance disadvantages, such as when invoking the degenerate expression on most inputs.

Reply:
I don't use walrus operator because of your backport repository. Reduced cognitive complexity of this function.

Reply:
FTR, walrus operator was introduced in Python 3.8, so is suitable for backport.

In the latest revision, I've refactored the comment handling in _CommentSpec to encapsulate it and unify the approach, with no `if` statements needed.